specs/models/Step.v1.json

```diff
@@ -0,0 +1,181 @@
+{
+  "title": "Step.v1",
+  "type": "object",
+  "properties": {
+    "options": {
+      "type": "object",
+      "properties": {
+        "sourceQuery": {
+          "type": "string",
+          "description": "Defines the items to be processed by the step; must be a cts.query or cts.uris statement if sourceQueryIsScript is false"
+        },
+        "sourceQueryIsScript": {
+          "type": "boolean",
+          "description": "Added in 5.3.0; if true, then sourceQuery can be any JavaScript statement that can be passed into xdmp.eval"
+        },
+        "constrainSourceQueryToJob": {
+          "type": "boolean",
+          "description": "If true, the query is applied to the documents that were created or modified in the same job that executes the step"
+        },
+        "provenanceGranularityLevel": {
+          "type": "string",
+          "description": "The granularity of the provenance tracking information: coarse (default) to store document-level provenance information only, fine to store document-level and property-level provenance information, or off to disable provenance tracking in future job runs. Applies only to mapping, matching, merging, mastering, and custom steps.",
+          "enum": [
+            "off",
+            "coarse",
+            "fine"
+          ]
+        },
+        "stepUpdate": {
+          "type": "boolean",
+          "description": "If true, custom modules can make changes directly to records in the database"
+        },
+        "acceptsBatch": {
+          "type": "boolean",
+          "description": "If true, the step module is invoked once with all records in the batch passed to it"
+        },
+        "collections": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "additionalCollections": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "permissions": {
+          "type": "string",
+          "description": "Comma-delimited string of role,capability,role,capability,etc"
+        },
+        "outputFormat": {
+          "type": "string",
+          "enum": [
+            "json",
+            "xml"
+          ]
+        },
+        "sourceDatabase": {
+          "type": "string"
+        },
+        "targetDatabase": {
+          "type": "string"
+        },
+        "targetEntity": {
```

Comment:
I think we should at a minimum say "targetEntityTypeId" here, as "entityTypeId" is our agreed-upon name for a unique identifier for an entity type. We can have code that accepts a name in here as well. 

Also, I'm proposing "inputDatabase" and "outputDatabase" to be consistent not just with mlcp, but also with the evolving concepts of a Step having an Input and an Output. Gorbet pointed out how the word "source" is a bit loaded because it has been typically used to refer to a source of data outside of ML, like some external system. So we want to move away from "sourceQuery" - though I wouldn't rename that yet. I don't know, if this is just a draft, I think we can stay with sourceDatabase/targetDatabase, but I think we'll need to get Gorbet and likely PAT involved because we're really defining critical concepts here.

Reply:
changed to "targetEntityTypeId" and made the changes to input/output database